File: app/src/admin/api/vision.ts
import { CommonResponse } from "@/api/common.ts";
import axios from "axios";
import { getErrorMessage } from "@/utils/base.ts";

export type VisionConfig = {
  models: string[];
  treat_all_as_vision?: boolean;
};

export type VisionResponse = CommonResponse & {
  data?: VisionConfig;
};

export async function getVisionConfig(): Promise<VisionResponse> {
  try {
    const response = await axios.get("/admin/vision/config");
    return response.data as VisionResponse;
  } catch (e) {
    return { status: false, error: getErrorMessage(e) };
  }
}

export async function updateVisionConfig(
  models: string[],
  treatAllAsVision?: boolean,
): Promise<CommonResponse> {
  try {
    const response = await axios.post("/admin/vision/config", {
      models,
      treat_all_as_vision: treatAllAsVision,
    });
    return response.data as CommonResponse;
  } catch (e) {
    return { status: false, error: getErrorMessage(e) };
  }
}

export async function refreshVisionConfig(): Promise<CommonResponse> {
  try {
    const response = await axios.post("/admin/vision/refresh");
    return response.data as CommonResponse;
  } catch (e) {
    return { status: false, error: getErrorMessage(e) };
  }
}

File: admin/vision.go
package admin

import (
	"chat/utils"
	"net/http"

	"github.com/gin-gonic/gin"
)

type VisionConfigForm struct {
	Models           []string `json:"models"`
	TreatAllAsVision bool     `json:"treat_all_as_vision"`
}

// GetVisionConfigAPI 获取视觉模型配置
func GetVisionConfigAPI(c *gin.Context) {
	cfg := utils.GetVisionConfig()
	c.JSON(http.StatusOK, gin.H{
		"status": true,
		"data":   cfg,
	})
}

// UpdateVisionConfigAPI 更新视觉模型配置
func UpdateVisionConfigAPI(c *gin.Context) {
	var form VisionConfigForm
	if err := c.ShouldBindJSON(&form); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{
			"status": false,
			"error":  err.Error(),
		})
		return
	}

	// 使用独立配置保存函数
	err := utils.UpdateVisionConfig(form.Models, form.TreatAllAsVision)

	c.JSON(http.StatusOK, gin.H{
		"status": err == nil,
		"error":  utils.GetError(err),
	})
}

// RefreshVisionConfigAPI 刷新视觉模型配置(使配置生效)
func RefreshVisionConfigAPI(c *gin.Context) {
	// 重新加载配置到内存
	cfg := utils.GetVisionConfig()
	utils.RefreshCustomVisionModels(cfg.Models)

	c.JSON(http.StatusOK, gin.H{
		"status":  true,
		"message": "Vision models configuration refreshed successfully",
	})
}
